### ISSUE TITLE
Thermostat (WK) HomeKit Cool setting not mapped to "cold" mode

### ISSUE THREAD

**Issue by KiwiBlueSpring** [0x5e/homebridge-tuya-platform#241]
Ive got a tuya thermostat device which has four different values for the "mode" dp ["cold", "hot", "wind," "dry"]

In homekit, Off & Heat work correctly, however Auto &  Cool dont work. Looking at HB-Debug logs, the plugin sends the command "eco" when pressing cool instead of "cold" for my use case. I did see this thread; #2  but I interpreted it as cool would work if the device had either "eco" or "cold" just encase.

Pressing Auto in homekit does nothing, no logs in homebridge.

On another note, my thermostat has fan speeds, ("low", "middle", "high", "auto")  im not sure if the homekit thermostat service supports this or not, or if I should try setting it to an air-conditioner that uses the heaterCooler service and mapping the data points. (not sure if thats possible or not)

**Homebridge Logs**

_Pressing Cool on HomeKit:_

[@0x5e/homebridge-tuya-platform] [TuyaOpenAPI] Request:
method = post
endpoint = https://openapi.tuyaus.com
path = /v1.0/devices/ebe271f6f7ab3699a19vhs/commands
query = null
headers = {
  "t": "1677063073776",
  "client_id": "r993gagtar6qw1586vcb",
  "nonce": "42d42522-641b-4c08-8fca-a68c807f20c7",
  "Signature-Headers": "client_id",
  "sign": "9CFBF89A2FD1450AA2C658F2476FFED89FF2A96EF65682C5796B36C1A1362F87",
  "sign_method": "HMAC-SHA256",
  "access_token": "f5587988051137f2c5025663d767d756",
  "lang": "en",
  "dev_lang": "javascript",
  "dev_channel": "homebridge",
  "devVersion": "1.7.0-beta.28"
}
body = {
  "commands": [
    {
      "code": "mode",
      "value": "eco"
    }
  ]
}
[@0x5e/homebridge-tuya-platform] [TuyaOpenAPI] Response:
path = /v1.0/devices/ebe271f6f7ab3699a19vhs/commands
data = {
  "code": 1101,
  "msg": "params range invalid",
  "success": false,
  "t": 1677063074527,
  "tid": "d3a55d73b29e11edb2d1c2a5aeef7bb2"
}

**Device info**

`"switch": "{true,false}"`

`"mode": {"range": [ "cold", "hot", "wind", "dry"]}`

`"temp_set": {"unit": "℃", "min": 16, "max": 32, "scale": 0, "step": 1}`

`level: {"range": ["low", "middle", "high", "auto"]}`

`child_lock: "{true,false}"`

**Comment by 0x5e** [comment]
Thanks for the feedback, I found the typo when processing eco and cold mode, next commit will fix the cold mode issue.
The auto mode should be hidden for your device, but there's many condition select code that I didn't remember why do this, I suggest just keep that logic to avoid other mistakes.

thermostat service doen't support fan speed, if you want this feature, you can override the device category to ac's category `kt`, and override `level` to `fan_speed_enum` according to product document (remember to clean accessory cache). speed auto is not supported, other speed level will be mapped to 0-100% according to their orders.

**Comment by 0x5e** [comment]
I've read the tuya documentation and some thermostat products (don't have experience using thermostat), so many of them have wind mode right? I can add this feature like ac when I'm free. always busy these days :(

**Comment by KiwiBlueSpring** [comment]
Thank you for fixing the cool mode issue so quickly! I look forward to the update. :) 

For fan speed I’m okay with just overriding the device category, however I’ve just gone ahead and tried that, and I get a ‘not supported’ accessory in HomeKit with the general house icon. Not sure if I’m doing something wrong?


This is the config
<img width="279" alt="CleanShot 2023-02-24 at 08 26 02@2x" src="https://user-images.githubusercontent.com/72439854/221010117-bf132499-00de-485d-92b5-5b958280efd4.png"> 

![10264A7E-AE2E-4645-B839-1872A3A42ED4](https://user-images.githubusercontent.com/72439854/221010245-fe3af686-cc4e-42a4-88d7-e95ab27ac285.jpeg)

**Comment by 0x5e** [comment]
try clean accessory cache

**Comment by KiwiBlueSpring** [comment]
Yup, this was after clearing the accessory cache. Ive done it again and still same issue.

**Comment by 0x5e** [comment]
are there any warning/error logs from homebridge?

**Comment by KiwiBlueSpring** [comment]
Thank you. I realised I can filter out the logs in portainer which made it 1000x easier to catch the error without all the other giant logs moving so quickly. It was missing a required schema:

_[Thermostat] Missing one of the required schema: [ 'temp_current' ]_

<img width="550" alt="CleanShot 2023-02-28 at 17 11 25@2x" src="https://user-images.githubusercontent.com/72439854/221751694-b7f84031-ae7d-4e8f-84cd-fca13b22ac5d.png">

I mapped it to 'temp_set' and seems to work fine so far. 

Both kt & ktkzq seems to expose a an actual fan accessory grouped with the ac device itself which didnt seem to do anything when adjusting. Once I clicked 'show as seperate tiles' the ac card updated and showed a fan speed slider that worked fine & left an extra fan tile outside. Just something interesting I found.

Ive attached some screenshots if ur interested 
_(1 =  Ac tile before ungroup | 2 =  ac tile after ungroup | 3 = ac & fan seperated cards)_

<div style="display: flex; flex-direction: row;">


<img width="45%" alt="Before Ungroup" src="https://user-images.githubusercontent.com/72439854/221754020-c3a2a0a3-16bb-4eba-86a9-841d005ffe89.PNG">




<img width="45%" alt="After Ungroup" src="https://user-images.githubusercontent.com/72439854/221754275-834c4386-7158-495e-9154-f1b3834c5bec.PNG">



<img width="90%" alt="The seperated tiles" src="https://user-images.githubusercontent.com/72439854/221754332-0d44fa75-eecf-4800-8d47-81a44861c8e5.jpeg">

</div>

-------




I have one other issue, I noticed that the devices current status doesnt seem to update in homekit / homebridge when there are adjustments made on the thermostat itself (or even adjustments made in the smartlife app). Like if I turn on the thermostat, it doesnt show it being 'on' in homekit, or if i adjust anything like temperature, it wont relfect those changes in homekit either.

After a plugin restart it seems to fetch the devices info and it becomes up to date. However any changes after that wont relfect. Is there any reason why this could be. There arent any logs that I can see that could point out an issue. 

Btw the 'cool' mode works great after the update so thank you so much & I am loving the plugin!
